Sources/GDBRemote/Mixins/FileOperationsMixin.hpp

```diff
@@ -130,18 +130,26 @@ ErrorCode FileOperationsMixin<T>::onQueryModuleInfo(Session &session,
                                                      std::string &triple,
                                                      ModuleInfo &info) const {
   ByteVector buildId;
-  if (Platform::GetExecutableFileBuildID(path, buildId)) {
-    // format the uuid as an upper-case string with two hex chars per byte
-    std::ostringstream ss;
-    for(const auto b : buildId)
-      ss << std::uppercase << std::hex << std::setfill('0') << std::setw(2) << int(b);
-
-    info.uuid = ss.str();
+  if (!Platform::GetExecutableFileBuildID(path, buildId)) {
+    // Not all executable files contain an embedded build ID. In this case,
+    // return a crc32 of the contents of the file. The documentation for
+    // `qModuleInfo` suggests an md5 hash can be returned as "md5" in the
+    // response instead of "uuid." However, returning a crc32 is consistent
+    // with behavior of lldb-server.
+    uint32_t crc;
+    CHK(File::crc32(path, crc));
```

Comment:
Technically, the computation is over the normative portions of the file, not the entire file. I suppose that there is the performance aspect - digesting the entire file is faster than parsing and processing the normative parts. Not doing the slower process means that we would not be able to correctly identify files, which leaves this partially incorrect.

Reply:
From what I've observed, the CRCs calculated this way exactly match those sent by lldb-server on the same files. Am I getting lucky? Is there any documentation on what parts of the file should be covered by the crc/hash?

Reply:
Documentation? Ha! The source code to the linker is the documentation 😢. IIRC, it was `.text`, `.data`, and any custom sections that are summed.